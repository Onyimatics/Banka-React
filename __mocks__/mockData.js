import jwt from 'jsonwebtoken';

const secret = 'onyinye';

export const userToken = `Bearer ${jwt.sign(
  {
    id: 1,
    email: 'example@gmail.com',
    role: 'user',
    status: 'verified'
  },
  secret,
  { expiresIn: '1 hour' }
)}`;

export const expiredToken = `Bearer ${jwt.sign(
  {
    id: 1,
    email: 'example@gmail.com',
    role: 'admin',
    status: 'verified'
  },
  secret,
  { expiresIn: -1 }
)}`;

export default {
  authResponse: {
    data: {
      id: 1,
      username: 'Onyimatics',
      email: 'onyimatics@andela.com',
      role: 'user',
      status: 'verified',
      token: userToken
    }
  }
};

export const mockStoreData = {
  allAccountsData: {
    data: {
      rows: [
        {
          Balance: 2000,
          accountNumber: 1102345718,
          createdOn: "2019-08-29T00:43:11.908Z",
          status: "active",
          type: "savings"
        }
      ],
      accounts: { type: 'savings', openingBalance: 1000 }
    }
  },
  newAccountData: {
    data: {
      articleId: 55,
      content: 'good',
      createdAt: '2019-08-26T05:14:27.383Z',
      id: 72,
      updatedAt: '2019-08-26T05:14:27.383Z',
      userId: 162
    }
  },
  errorData: {
    message: 'Unable to create account'
  }
};
