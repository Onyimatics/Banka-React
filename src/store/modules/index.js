import { combineReducers } from 'redux';
import authReducer from './auth/reducer';
import accountReducer from './accounts/reducer';
import transactionReducer from './transactions/reducer';

const rootReducer = combineReducers({
  authReducer,
  accountReducer,
  transactionReducer
});
export default rootReducer;
