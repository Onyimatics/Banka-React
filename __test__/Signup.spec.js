import React from 'react';
import { mount, shallow } from './enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import { rootReducer } from '../src/store/modules/index';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
// import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import SignUpForm from '../src/components/Signup';

import { userToken } from '.././__mocks__/mockData'


const createStore = (isAuthenticated = true, user = {}, status = 'rest') => {
  const content = {
    authReducer: {
      isAuthenticated: true,
      user,
      status: 'authenticationSuccess'
    }
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const store = mockStore({
    authReducer: { ...content, isAuthenticated, user, status }
  });
  return store;
};

const props = {
  authAction: jest.fn(),
  status: 'rest',
  isAuthenticated: false,
  error: null,
  location: { url: '/user-dashboard' },
  history: { push: jest.fn() }
};


describe('Signup component', () => {
  beforeEach(() => {
    location = {
      search: `token=${userToken}`,
      url: '/'
    };
  });
  it('renders without crashing', () => {
    const wrapper = mount(
      <Provider store={createStore()}>
        <Router>
         <Switch>
           
            <Route SignUpForm {...props} /> 
          </Switch>
        </Router>
      </Provider>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders one form', () => {
    const wrapper = mount(
      <Provider store={createStore()}>
        <Router>
          <SignUpForm {...props} location={location} />;
        </Router>
      </Provider>
    );



    expect(wrapper.find('h2').text()).toEqual('Sign Up');
    expect(wrapper.find('SignUpForm')).toBeTruthy();
    expect(wrapper.find('form')).toBeTruthy();
    expect(wrapper.find('form').length).toBe(1);
  });

  it('renders 4 input fields and 1 submit button ', () => {
    const wrapper = mount(
      <Provider store={createStore()}>
        <Router>
      
          <SignUpForm {...props} location={location} /> 
         
        </Router>
      </Provider>
    );

    expect(wrapper.find('input').length).toBe(4);
    expect(wrapper.find('input').at(0).props().type).toBe('text');
    expect(wrapper.find('input').at(1).props().type).toBe('text');
    expect(wrapper.find('input').at(2).props().type).toBe('email');
    expect(wrapper.find('input').at(3).props().type).toBe('password');
    expect(wrapper.find('button').length).toBe(2);
    expect(wrapper.find('button').at(0).props().type).toBe(undefined);
    expect(wrapper.find('button').at(1).props().type).toBe('submit');
  });
})
