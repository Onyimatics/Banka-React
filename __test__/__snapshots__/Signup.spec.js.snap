// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Switch>
        <Route
          SignUpForm={true}
          authAction={[MockFunction]}
          computedMatch={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          error={null}
          history={
            Object {
              "push": [MockFunction],
            }
          }
          isAuthenticated={false}
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          status="rest"
        />
      </Switch>
    </Router>
  </BrowserRouter>
</Provider>
`;
